backend/app/Actions/User/UpdateUserAction.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Actions\User;
+
+use App\Exceptions\UserNotFoundException;
+use App\Repositories\Contracts\UserRepository;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+
+final class UpdateUserAction
+{
+    private $userRepository;
+
+    public function __construct(UserRepository $userRepository)
+    {
+        $this->userRepository = $userRepository;
+    }
+
+    public function execute(UpdateUserRequest $request): UpdateUserResponse
+    {
+        try {
+            $user = $this->userRepository->getById($request->id());
+        } catch (ModelNotFoundException $exception) {
+            throw new UserNotFoundException();
+        }
+
+        $user->email = $request->email() ?: $user->email;
```

Comment:
add to your getters default value and pass there $user->email, it reads much better

public function getEmail(string $default = '') : string { return $this->email ?: $default; }

Reply:
$request->getEmail($user->email)